modules/capi-yaml-machine-set-vsphere.adoc

```diff
@@ -0,0 +1,78 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cluster_api_machine_management/cluster_api_provider_configurations/cluster-api-config-options-vsphere.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="capi-yaml-machine-set-vsphere_{context}"]
+= Sample YAML for a Cluster API compute machine set resource on {vmw-full}
+
+The compute machine set resource defines additional properties of the machines that it creates.
+The compute machine set also references the infrastructure resource and machine template when creating machines.
+
+[source,yaml]
+----
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineSet
+metadata:
+  name: <machine_set_name> # <1>
+  namespace: openshift-cluster-api
+spec:
+  clusterName: <cluster_name> # <2>
+  replicas: 1
+  selector:
+    matchLabels:
+      test: example
+  template:
+    metadata:
+      labels:
+        test: example
+    spec:
+      bootstrap:
+         dataSecretName: worker-user-data # <3>
+      clusterName: <cluster_name>
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: VSphereMachineTemplate # <4>
+        name: <template_name> # <5>
+      failureDomains: # <6>
```

Comment:
Do we do this also for other providers?
To encourage MSets to be on different FDs?

Reply:
Yeah I am not sure what the best approach is for these. If we want to encourage multiple, then I think we should probably show it in the sample even if that makes it a bit long. Is multiple a best practice?

Reply:
Multiple FDs are not available in a MAPI MachineSet.
I'd probably go the single FD route for the CAPI examples.

cc. @JoelSpeed

Reply:
Yeah I think I'd show only a single failure domain use case here right now at least. I don't believe we've tested using multiple failure domains in CAPI yet so we don't know the implications of this on other areas of the product. Documenting it makes it supported and I'd prefer to test it before we do so

Reply:
That all sounds like solid reasoning to me :)